IdentityCore/tests/MSIDBoundRefreshTokenRedemptionTests.m

```diff
@@ -0,0 +1,1121 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+#if TARGET_OS_IPHONE
+#import <XCTest/XCTest.h>
+#import "MSIDBoundRefreshToken.h"
+#import "MSIDBoundRefreshToken+Redemption.h"
+#import "MSIDBoundRefreshTokenRedemptionParameters.h"
+#import "MSIDWPJKeyPairWithCert.h"
+#import "MSIDEcdhApv.h"
+#import "MSIDJWECrypto.h"
+#import "MSIDJWTHelper.h"
+#import "MSIDAccountIdentifier.h"
+#import "MSIDTestSecureEnclaveKeyPairGenerator.h"
+#import "MSIDWPJKeyPairWithCert+TransportKey.h"
+#import "MSIDWorkPlaceJoinConstants.h"
+#import "MSIDKeychainUtil.h"
+#import "NSData+MSIDExtensions.h"
+#import "MSIDTestSwizzle.h"
+#import "MSIDWorkPlaceJoinUtil.h"
+
+@interface MSIDBoundRefreshTokenRedemptionTests : XCTestCase
+@property (nonatomic) MSIDTestSecureEnclaveKeyPairGenerator *deviceKeyGenerator;
+@property (nonatomic) MSIDTestSecureEnclaveKeyPairGenerator *transportKeyGenerator;
+@property (nonatomic) NSString *tenantId;
+@property (nonatomic) NSString *deviceKeyTag;
+@property (nonatomic) NSString *transportKeyTag;
+@property (nonatomic) NSString *accessGroup;
+@property (nonatomic) NSString *deviceId;
+@end
+
+static const NSString *kAuthorityUrl = @"https://login.microsoftonline.com/common/oauth2/v2.0/token";
+
+@implementation MSIDBoundRefreshTokenRedemptionTests
+
+- (void)setUp
+{
+    [super setUp];
+    [self cleanUpWpjInformation];
+    self.tenantId = NSUUID.UUID.UUIDString;
+    self.accessGroup = [NSString stringWithFormat:@"%@.com.microsoft.workplacejoin.v2", [[MSIDKeychainUtil sharedInstance] teamId]];
```

Comment:
Do we run these tests on Mac? if yes, will we mess up our prod keychain?  Can we mock keychain logic or at least use "test" accessGroup instead?